src/lib/routing-tables.js

```diff
@@ -97,18 +112,59 @@ class RoutingTables {
   }
 
   _removeRoute (ledgerB, ledgerC, connectorFromBToC) {
+    let lostLedgerLinks = []
     this.eachSource((tableFromA, ledgerA) => {
       if (ledgerA !== ledgerB) return
-      tableFromA.removeRoute(ledgerC, connectorFromBToC)
+      if (tableFromA.removeRoute(ledgerC, connectorFromBToC)) {
+        lostLedgerLinks.push(ledgerA)
+      }
     })
+    return lostLedgerLinks
   }
 
   removeExpiredRoutes () {
+    let lostLedgerLinks = []
     this.eachRoute((routeFromAToB, ledgerA, ledgerB, nextHop) => {
       if (routeFromAToB.isExpired()) {
-        this._removeRoute(ledgerA, ledgerB, nextHop)
+        debug('removing expired route ledgerA:', ledgerA, ' ledgerB:', ledgerB, ' nextHop:', nextHop)
+        let lll = this._removeRoute(ledgerA, ledgerB, nextHop)
+        lostLedgerLinks.concat(lll)
```

Comment:
This is a no-op. `concat` is non-destructive and returns the concatenated array.